src/controller/CHIPDeviceController.cpp

```diff
@@ -1727,46 +1728,84 @@ void DeviceCommissioner::OnDone()
 void DeviceCommissioner::OnArmFailSafe(void * context,
                                        const GeneralCommissioning::Commands::ArmFailSafeResponse::DecodableType & data)
 {
-    // TODO: Use errorCode
-    ChipLogProgress(Controller, "Received ArmFailSafe response");
+    CommissioningDelegate::CommissioningReport report;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    ChipLogProgress(Controller, "Received ArmFailSafe response errorCode=%u", static_cast<uint8_t>(data.errorCode));
+    if (data.errorCode != GeneralCommissioning::CommissioningError::kOk)
+    {
+        err = CHIP_ERROR_INTERNAL;
```

Comment:
What we currently do for some of the operational credentials errors is encode them in CHIP_ERROR values.

I really wish these clusters had just used cluster-specific status to indicate errors, because then we could have stuck that into the CHIP_ERROR easily.  But maybe as a followup (separate PR) we should consider having some sort of "cluster replied sith a status enum, here is the value" setup for CHIP_ERROR, so that the error we set here could be meaningful even if we don't have the full report?

Reply:
If I am understanding correctly, you are asking for some sort of mapping where based on some sort of `GeneralCommissioning::CommissioningError` value that is not `kOk` that we give a more appropriate CHIP_ERROR.

If so I did this earlier on in over here: https://github.com/project-chip/connectedhomeip/pull/17778/commits/1852ff276f77ad747d25957b1ed65f1302022cc9#diff-30c3779ffb95f404ea48b1ed6319a810a1f5639a9b035002f11a44193b4cddbeR1729

I can bring that back if you'd like with more appropriate errors

Reply:
I think it would be a good idea, but with the CHIP_ERROR encoding the actual status, not via making up various new CHIP_ERROR values..... Separate work from this PR, as I said.